### PR TITLE
BigInt and BigNum are inheriting Number class

### PR DESCRIPTION
**Pull request by KamilKurde**
for easier conversion between numeric types

### CODE REVIEW

File: kbignum/src/commonMain/kotlin/com/soywiz/kbignum/BigNum.kt [hunk 1]
```diff
@@ -31,6 +31,7 @@ class BigNum(val int: BigInt, val scale: Int): Number() {
         else -> BigNum(int / (10.bi pow (this.scale - otherScale)), otherScale)
     }
 
+    operator fun unaryMinus(): BigNum = this * BigNum("-1")
```
Comment (soywiz) [inline kbignum/src/commonMain/kotlin/com/soywiz/kbignum/BigNum.kt]: What about `BigNum(-int, scale)`? That should be faster I guess

https://github.com/korlibs/kbignum/blob/cf76731478c8fbc95dc6e7c3fed6b19a4d4963ca/kbignum/src/commonMain/kotlin/com/soywiz/kbignum/BigInt.kt#L318

Comment (soywiz) [reply kbignum/src/commonMain/kotlin/com/soywiz/kbignum/BigNum.kt]: Can you add a test here verifying the new `unaryMinus`?

https://github.com/korlibs/kbignum/blob/52cd24dab727fb56c29039e3533a15d70025898e/kbignum/src/commonTest/kotlin/com/soywiz/kbignum/BigNumTest.kt#L7

Comment (KamilKurde) [reply kbignum/src/commonMain/kotlin/com/soywiz/kbignum/BigNum.kt]: Sure, both suggestions applied.

**Comment by KamilKurde** [comment]
It looks like internal Kotlin/JS compiler error
>  Task :kbignum:jsLegacyTestClasses
  Uncaught TypeError: Cannot read property 'prototype' of undefined
>
> Task :kbignum:jsLegacyBrowserTest
  at /home/runner/work/kbignum/kbignum/build/js/packages/kbignum-root-kbignum-jsLegacy-test/adapter-browser.js:65825:45
SourceMap position not found for trace: Uncaught TypeError: Cannot read property 'prototype' of undefined
  TypeError: Cannot read property 'prototype' of undefined
at http://localhost:9876/absolute/home/runner/work/kbignum/kbignum/build/js/packages/kbignum-root-kbignum-jsLegacy-test/adapter-browser.js?a38989c4f150df29f6768d8067d26af748b7407b:65825:45
      at Object.<anonymous> (/home/runner/work/kbignum/kbignum/build/js/packages/kbignum-root-kbignum-jsLegacy-test/adapter-browser.js:65825:45)

**Comment by soywiz** [comment]
Yeah. I wanted to drop support for the legacy compiler, but some people is still using it, and sadly this would break the code for that people. Maybe we can make an intermediate approach? Just removing the extending Number(), and providing the `to*`() methods and a `toNumber()` method? Then create an issue to reenable extending Number once either IR is stable, or the issue is fixed on legacy

**Comment by soywiz** [comment]
I have created this issue: https://youtrack.jetbrains.com/issue/KT-47393

**Comment by KamilKurde** [comment]
>and providing the `to*`() methods and a `toNumber()` method? 

how would it work? Number is an abstract class.

**Comment by soywiz** [comment]
You are right! Then just the `to*` methods without anything extra

**Pull request closed without merging**